test/sharness/t0010-basic-commands.sh

```diff
@@ -102,6 +102,19 @@ test_expect_failure "All ipfs commands docs are 80 columns or less" '
   fi
 '
 
+test_expect_success "All ipfs commands fail when passed a bad flag" '
+  echo 0 > fail
+  while read -r cmd
+  do
+    test_must_fail $cmd --badflag </dev/null >/dev/null ||
+      { echo $cmd exit with code 0 when passed --badflag; echo 1 > fail; }
```

Comment:
Should definitely quote this. Also, we can probably just touch fail and check to see if it exists (simpler).

Reply:
About just touching fail, if we do that we then have to make sure that we remove fail in case of failure so that later tests that use the same mechanism don't fail just because of the leftover file.

Reply:
About quoting I tried to follow the same pattern as other tests. I will take a look at quoting everywhere.

Reply:
Meh. Let's just leave the failure mechanism as-is.